pandemic51/core/streaming.py

```diff
@@ -245,35 +111,420 @@ def sample_first_frame(inpath, outpath, width=None, height=None):
     return True
 
 
-def download_and_store(stream_name, outdir, width=None, height=None):
-    '''Downloads an image from the latest stream, and add it to the database.
+class Stream(etas.Serializable):
+    '''Abstract class for streams of various types providing functionality for
+    downloading clips and images from the stream, getting the latest stream URL,
+    etc.
+    '''
+    def __init__(self, stream_name, GMT):
+        '''
+        Args:
+            stream_name: name of the stream (should match JSON file in
+                {{pandemic51}}/config/streams/
+            GMT: integer relative timezone of stream
+        '''
+        self.type = etau.get_class_name(self)
+        self.stream_name = stream_name
+        self.GMT = GMT
+
+    @staticmethod
+    def stream_path(stream_name):
+        '''Get the path to the Stream JSON file.
+
+        Args:
+            stream_name: name of the stream
+
+        Returns:
+            path the the Stream object serialized on disk
+        '''
+        return os.path.join(panc.STREAMS_DIR, stream_name + ".json")
+
+    @staticmethod
+    def get_stream_names():
```

Comment:
convert to a public module method? It's about all streams, not one particular stream